include/alpaka/kernel/TaskKernelGpuUniformCudaHipRt.hpp

```diff
@@ -390,6 +468,315 @@ namespace alpaka
+                {
+                    throw std::runtime_error(
+                        "The given work division is not valid or not supported by the device of type "
+                        + getAccName<AccGpuUniformCudaHipRt<TApi, TDim, TIdx>>() + "!");
+                }
+#        endif
+
+                // Get the size of the block shared dynamic memory.
+                auto const blockSharedMemDynSizeBytes = std::apply(
+                    [&](remove_restrict_t<std::decay_t<TArgs>> const&... args) {
+                        return getBlockSharedMemDynSizeBytes<TAcc>(
+                            task.m_kernelFnObj,
+                            blockThreadExtent,
+                            threadElemExtent,
+                            args...);
+                    },
+                    task.m_args);
+
+#        if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
+                // Log the block shared memory idx.
+                std::cout << __func__ << " BlockSharedMemDynSizeBytes: " << blockSharedMemDynSizeBytes << " B"
+                          << std::endl;
+#        endif
+                auto kernelName = alpaka::detail::
+                    gpuKernel<TKernelFnObj, TApi, TAcc, TDim, TIdx, remove_restrict_t<std::decay_t<TArgs>>...>;
+
+#        if ALPAKA_DEBUG >= ALPAKA_DEBUG_MINIMAL
+                {
+                    // This checks if requested number of blocks is compliant with the maxima of the accelerator.
+                    int numBlocksPerSm = 0;
+                    cudaOccupancyMaxActiveBlocksPerMultiprocessor(
+                        &numBlocksPerSm,
+                        kernelName,
+                        blockThreadExtent.prod(),
+                        static_cast<std::size_t>(blockSharedMemDynSizeBytes));
+
+                    if(numBlocksPerSm
+                           * trait::GetAccDevProps<TAcc>::getAccDevProps(getDev(queue)).m_multiProcessorCount
+                       < gridBlockExtent.prod())
+                    {
+                        throw std::runtime_error(
+                            "The number of requested blocks is larger than maximuma of the device for the kernel "
+                            + core::demangled<TKernelFnObj>
+                            + "! Device: " + getAccName<AccGpuUniformCudaHipRt<TApi, TDim, TIdx>>()
+                            + ", requested: " + std::to_string(gridBlockExtent.prod()) + ", maximum allowed: "
+                            + std::to_string(
+                                trait::GetAccDevProps<TAcc>::getAccDevProps(getDev(queue)).m_multiProcessorCount)
+                            + ".");
+                    }
+                }
+
+#        endif
+
+#        if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
+                {
+                    // Log maximum number of blocks that can be requested by this kernel.
+                    int numBlocksPerSm = 0;
+                    cudaOccupancyMaxActiveBlocksPerMultiprocessor(
```

Comment:
could you store the result and reuse it, instead of calling this twice ?

Reply:
Of course, I thought it would be a little bit more clear, considering these are 2 different debug levels

Reply:
I see.

Given that one is always a subset of the other, you could nest them:
```c++
#        if ALPAKA_DEBUG >= ALPAKA_DEBUG_MINIMAL 
    // here goes the initialisation of the variables common between full and minimal
    // and the code for minimal debug
    ...
#            if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
    // here goes the code for full debug
    ...
#            endif
#        endif
```